### PR TITLE
feat: Implement an endpoint to add a new vocabulary entry

### PR DESCRIPTION
**Pull request by ArzamastsevOleksandr**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version: '3.8'
+services:
+  db:
+    image: postgres:14.1-alpine
+#    restart: always
+    environment:
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=pass
+    ports:
+      - '5432:5432'
```
Comment (LarysaDoroshenko) [inline docker-compose.yml:10]: Change, please, mapping DB port 5432 to 5455, for example. As another container already uses this port.

File: src/main/java/com/booster/server/CreateVocabularyEntryInput.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.booster.server;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/CreateVocabularyEntryInput.java:6]: I think, @Data aggregates more annotations than you currently need. Remove @Data and add only necessary annotations.

File: src/main/java/com/booster/server/CreateVocabularyEntryInput.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+@Builder
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/CreateVocabularyEntryInput.java:7]: I think @Builder generates unnecessary code (you can see it in the compiled class version). The same functionality can be reached with @Setter and @Accessors(chain=true) annotations.

File: src/main/java/com/booster/server/VocabularyEntryDto.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.booster.server;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/VocabularyEntryDto.java:7]: The same suggestions are here.

File: src/main/java/com/booster/server/VocabularyEntryRepository.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.booster.server;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (LarysaDoroshenko) [inline src/main/java/com/booster/server/VocabularyEntryRepository.java:6]: You can omit @Repository annotation as you extend JpaRepository. Spring will create this repository anyway, but it depends on code style and preferences.

File: src/test/java/com/booster/server/VocabularyEntryControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Autowired
+    MockMvc mockMvc;
+    @Autowired
+    ObjectMapper objectMapper;
+    @Autowired
+    VocabularyEntryRepository vocabularyEntryRepository;
+
+    @Test
+    void create() throws Exception {
```
Comment (LarysaDoroshenko) [inline src/test/java/com/booster/server/VocabularyEntryControllerTest.java:30]: The name of the test is not describable. \
Also, this test will not work without the running DB container. Can we change this?

File: src/test/java/com/booster/server/VocabularyEntryControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        assertThat(vocabularyEntryRepository.findAll()).isEmpty();
+
+        mockMvc.perform(MockMvcRequestBuilders.post("/vocabulary-entry")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(objectMapper.writeValueAsString(CreateVocabularyEntryInput.builder()
+                                .name("a")
+                                .description("d")
+                                .build())))
```
Comment (LarysaDoroshenko) [inline src/test/java/com/booster/server/VocabularyEntryControllerTest.java:38]: Let`s create a test with a case of business logic, not just with random values.

File: docker-compose.yml (hunk 1 above)
Comment (ArzamastsevOleksandr) [reply docker-compose.yml:10]: Done

File: src/main/java/com/booster/server/CreateVocabularyEntryInput.java (hunk 2 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/CreateVocabularyEntryInput.java:6]: Done

File: src/main/java/com/booster/server/CreateVocabularyEntryInput.java (hunk 3 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/CreateVocabularyEntryInput.java:7]: Agreed. Fixed

File: src/main/java/com/booster/server/VocabularyEntryDto.java (hunk 4 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/VocabularyEntryDto.java:7]: Fixed

File: src/main/java/com/booster/server/VocabularyEntryRepository.java (hunk 5 above)
Comment (ArzamastsevOleksandr) [reply src/main/java/com/booster/server/VocabularyEntryRepository.java:6]: Agree, less but better

File: src/test/java/com/booster/server/VocabularyEntryControllerTest.java (hunk 7 above)
Comment (ArzamastsevOleksandr) [reply src/test/java/com/booster/server/VocabularyEntryControllerTest.java:38]: Fixed

File: src/test/java/com/booster/server/VocabularyEntryControllerTest.java (hunk 6 above)
Comment (ArzamastsevOleksandr) [reply src/test/java/com/booster/server/VocabularyEntryControllerTest.java:30]: Test renamed.

Added testcontainers support

**Review by LarysaDoroshenko — APPROVED** [review]

**Pull request merged**